bigtable-hbase-1.x-parent/bigtable-hbase-1.x-benchmarks/src/main/java/com/google/cloud/bigtable/hbase1_x/BigtableBenchmark.java

```diff
@@ -0,0 +1,211 @@
+import org.apache.hadoop.hbase.client.Connection;
+import org.apache.hadoop.hbase.client.Get;
+import org.apache.hadoop.hbase.client.Put;
+import org.apache.hadoop.hbase.client.Result;
+import org.apache.hadoop.hbase.client.ResultScanner;
+import org.apache.hadoop.hbase.client.Scan;
+import org.apache.hadoop.hbase.client.Table;
+import org.apache.hadoop.hbase.util.Bytes;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.infra.Blackhole;
+
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@Fork(value = 1)
+@Warmup(iterations = 10)
+@Measurement(iterations = 5)
+@State(Scope.Benchmark)
+public class BigtableBenchmark {
+
+  static final byte[] COL_FAMILY = "bm-test-cf".getBytes();
+  static final String SUFFIX_FMT = "%010d";
+  private static final long TOTAL_DATA_IN_BYTES = 1024 * 1024 * 1024;
+  private static final String BENCHMARK_WRITE_TABLE = "benchmark_writes";
+  private static final Pattern CELL_PATTERN = Pattern.compile("cellsPerRow/(\\d+)/cellSize/(\\d+)");
+
+  // TODO(rahulkql): Discuss and finalize If this actually needed.
+  static final int ROWS_PER_BATCH = 10000;
+
+  @Param("")
+  private String projectId;
+
+  @Param("")
+  private String instanceId;
+
+  @Param({
+    "cellsPerRow/10/cellSize/100",
+    "cellsPerRow/1/cellSize/1024",
+    "cellsPerRow/100/cellSize/1"
+  })
+  private String benchmarkConfig;
+
+  @Param({"true", "false"})
+  private boolean useGcj;
+
+  @Param("false")
+  private boolean useBatch;
+
+  private Connection connection;
+
+  private BenchmarkJobConfig jobConfig;
+
+  @Setup
+  public void setUp() throws IOException {
+    connection = BenchmarkDataUtils.createConnection(projectId, instanceId, useBatch, useGcj);
+    jobConfig = new BenchmarkJobConfig(benchmarkConfig);
+
+    createAndFillTable(connection, jobConfig);
+    createTableToWrite(connection, TableName.valueOf(BENCHMARK_WRITE_TABLE));
+  }
+
+  @Benchmark
+  public void pointRead(Blackhole blackhole) throws IOException {
+    Table table = connection.getTable(jobConfig.tableName);
+    String rowPrefix = getRowPrefix(jobConfig.cellsPerRow, jobConfig.cellSize);
+    Result result = table.get(new Get(Bytes.toBytes(rowPrefix + String.format(SUFFIX_FMT, 1))));
+
+    for (Cell cell : result.listCells()) {
+      blackhole.consume(cell);
+    }
+  }
+
+  @Benchmark
+  public void bulkScans(Blackhole blackhole) throws IOException {
+    String rowPrefix = getRowPrefix(jobConfig.cellsPerRow, jobConfig.cellSize);
```

Comment:
I don't think row keys need to be related to shape...each dataShape should have its own table